agent/functional_tests/tests/functionaltests_test.go

```diff
@@ -422,6 +423,42 @@ func TestCustomAttributesWithMaxOptions(t *testing.T) {
 	assert.True(t, ok, "OS attribute not found")
 }
 
+func waitForContainerHealthStatus(t *testing.T, testTask *TestTask) {
+	ctx, _ := context.WithTimeout(context.TODO(), waitTaskStateChangeDuration)
+waitHealth:
+	for {
+		select {
+		case <-ctx.Done():
+			t.Error("Waiting for container health status available timed out")
+		default:
+			testTask.Redescribe()
+			if aws.StringValue(testTask.Containers[0].HealthStatus) == "UNKNOWN" {
+				time.Sleep(time.Second)
+				continue
+			}
+			break waitHealth
+		}
+	}
+}
+
+// containerHealthMetricsTest tests the container health metrics based on the task definition
+func containerHealthMetricsTest(t *testing.T, taskDefinition string) {
```

Comment:
Is there a Docker version that's required for this test?

Reply:
Thanks for pointing that out, I have splited the test to also test docker that doesn't support startperiod.